crates/sui-surfer/src/default_surf_strategy.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use move_binary_format::normalized::Type;
+use move_core_types::language_storage::StructTag;
+use rand::{seq::SliceRandom, Rng};
+use sui_types::{
+    base_types::ObjectRef,
+    messages::{CallArg, ObjectArg},
+};
+use tokio::sync::watch;
+use tracing::debug;
+
+use crate::{
+    surf_strategy::SurfStrategy,
+    surfer_state::{EntryFunction, SurferState},
+};
+
+enum InputObjectPassKind {
+    Value,
+    ByRef,
+    MutRef,
+}
+
+#[derive(Default)]
+pub struct DefaultSurfStrategy {}
+
+#[async_trait]
+impl SurfStrategy for DefaultSurfStrategy {
+    async fn surf_for_a_while(
+        &mut self,
+        state: &mut SurferState,
+        mut entry_functions: Vec<EntryFunction>,
+        exit: &watch::Receiver<()>,
+    ) {
+        entry_functions.shuffle(&mut state.rng);
+        for entry in entry_functions {
+            let Some(args) = Self::choose_function_call_args(state, entry.parameters).await else {
+                debug!("Failed to choose arguments for Move function {:?}::{:?}", entry.module, entry.function);
+                continue;
+            };
+            state
+                .execute_move_transaction(entry.package, entry.module, entry.function, args)
+                .await;
+            if exit.has_changed().unwrap() {
+                return;
+            }
+        }
+    }
+}
+
+impl DefaultSurfStrategy {
+    async fn choose_function_call_args(
+        state: &mut SurferState,
+        params: Vec<Type>,
+    ) -> Option<Vec<CallArg>> {
+        let mut args = vec![];
+        let mut chosen_owned_objects = vec![];
+        let mut failed = false;
+        for param in params {
+            let arg = match param {
+                Type::Bool => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<bool>()).unwrap()),
+                Type::U8 => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<u8>()).unwrap()),
+                Type::U16 => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<u16>()).unwrap()),
+                Type::U32 => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<u32>()).unwrap()),
+                Type::U64 => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<u64>()).unwrap()),
+                Type::U128 => CallArg::Pure(bcs::to_bytes(&state.rng.gen::<u128>()).unwrap()),
+                Type::Address => CallArg::Pure(
+                    bcs::to_bytes(&state.cluster.accounts.choose(&mut state.rng)).unwrap(),
```

Comment:
Any reason why this has to be one of the accounts we know about? Maybe we can add some possibility of choosing a random address that's not one of the `cluster.accounts`.

Reply:
Probably, but that needs to happen very infrequently, because we won't be able to manipulate objects owned by those addresses.